File: 441.ArrangingCoins/coins.py
from math import sqrt

class Solution(object):
    def arrangeCoins(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans = int(sqrt(n * 2))
        return ans if ans * (ans + 1) / 2 <= n else ans - 1

if __name__ == '__main__':
    print Solution().arrangeCoins(1804289383)

File: 441.ArrangingCoins/Coins.java
class Solution {
    public int arrangeCoins(int n) {
        int ans = (int)(Math.sqrt(n) * Math.sqrt(2));
        return (int)((long)ans * (long)(ans + 1) / 2) <= n ? ans : ans - 1;
    }

    public static void main(String[] args) {
        System.out.println(new Solution().arrangeCoins(1804289383));
    }
}
